language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -313,8 +258,8 @@ tyEnvAssign (Unique i) ty = modify (over uniqueLookup (IM.insert i ty))
 -- if we are working with normalized type annotations
 normalizeTypeOpt :: Type TyNameWithKind () -> TypeCheckM a (NormalizedType TyNameWithKind ())
 normalizeTypeOpt ty = do
-    TypeConfig norm _ <- ask
-    if norm
+    typeConfig <- ask
```

Comment:
Why not use a pattern match?

Reply:
Because if we extend `TypeConfig` with a new field, we won't need to change anything here this way.